NavigationReactNative/src/android/app/src/main/java/com/navigation/reactnative/NavigationBarManager.java

```diff
@@ -0,0 +1,124 @@
+package com.navigation.reactnative;
+
+import android.graphics.Color;
+import android.os.Build;
+import android.view.MenuItem;
+import android.view.View;
+
+import com.facebook.react.bridge.ReadableArray;
+import com.facebook.react.bridge.ReadableMap;
+import com.facebook.react.common.MapBuilder;
+import com.facebook.react.uimanager.ThemedReactContext;
+import com.facebook.react.uimanager.ViewGroupManager;
+import com.facebook.react.uimanager.annotations.ReactProp;
+
+import java.util.Map;
+
+import javax.annotation.Nonnull;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.widget.Toolbar;
+
+public class NavigationBarManager extends ViewGroupManager<NavigationBarView> {
+    @Nonnull
+    @Override
+    public String getName() {
+        return "NVNavigationBar";
+    }
+
+    @Nonnull
+    @Override
+    protected NavigationBarView createViewInstance(@Nonnull ThemedReactContext reactContext) {
+        return new NavigationBarView(reactContext);
+    }
+
+    @Override
+    public int getChildCount(NavigationBarView parent) {
+        return parent.toolbar.getChildCount();
+    }
+
+    @Override
+    public View getChildAt(NavigationBarView parent, int index) {
+        return parent.toolbar.getChildAt(index);
+    }
+
+    @Override
+    public void addView(NavigationBarView parent, View child, int index) {
+        Toolbar.LayoutParams layoutParams = new Toolbar.LayoutParams(
+                Toolbar.LayoutParams.MATCH_PARENT,
+                Toolbar.LayoutParams.MATCH_PARENT
+        );
+        parent.toolbar.addView(child, index, layoutParams);
+        parent.toolbar.setContentInsetsRelative(0, 0);
+    }
+
+    @Override
+    public void removeViewAt(NavigationBarView parent, int index) {
+        parent.toolbar.removeViewAt(index);
+        parent.toolbar.setContentInsetsRelative(parent.defaultContentInsetStart, parent.defaultContentInsetEnd);
+    }
+
+    @ReactProp(name = "title")
+    public void setTitle(NavigationBarView view, @Nullable String title) {
+        view.setTitle(title);
+    }
+
+    @ReactProp(name = "logo")
+    public void setLogo(NavigationBarView view, @Nullable ReadableMap logo) {
+        view.setLogoSource(logo);
+    }
+
+    @ReactProp(name = "navigationImage")
+    public void setNavIcon(NavigationBarView view, @Nullable ReadableMap navIcon) {
+        view.setNavIconSource(navIcon);
+    }
+
+    @ReactProp(name = "overflowImage")
+    public void setOverflowIcon(NavigationBarView view, @Nullable ReadableMap overflowIcon) {
+        view.setOverflowIconSource(overflowIcon);
+    }
```

Comment:
When the `overflowImage` prop is null then this removes the default overflow icon. We need to store the `defaultOverflowIcon` in the View constructor and restored it when null